stix_shifter_modules/cybereason/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,471 @@
+from stix_shifter_utils.stix_translation.src.patterns.pattern_objects import ObservationExpression, \
+    ComparisonExpression, ComparisonExpressionOperators, ComparisonComparators, Pattern, \
+    CombinedComparisonExpression, CombinedObservationExpression, ObservationOperators
+import re
+import json
+from datetime import datetime, timedelta
+from os import path
+import copy
+from stix_shifter_utils.utils import logger
+
+logger = logger.set_logger(__name__)
+
+START_STOP_PATTERN = r"(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?Z)"
+MAC = '^(([0-9a-fA-F]{2}[:-]){5}([0-9a-fA-F]{2}))$'
+EMAIL = r'^\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b$'
+CONFIG_MAP_PATH = "json/config_map.json"
+
+STOP_TIME = datetime.utcnow()
+
+
+class LinkNotFoundException(Exception):
+    pass
+
+
+class QueryStringPatternTranslator:
+    """
+    comparator values to match with supported data source operators
+    """
+    comparator_lookup = {
+        ComparisonExpressionOperators.And: "AND",  # No 'AND' operator,
+                                                   # but implemented using filters and connectionFeature
+        ComparisonExpressionOperators.Or: "",  # 'OR' Not supported, error will be generated
+        ComparisonComparators.GreaterThan: "GreaterThan",
+        ComparisonComparators.GreaterThanOrEqual: "GreaterOrEqualsTo",
+        ComparisonComparators.LessThan: "LessThan",
+        ComparisonComparators.LessThanOrEqual: "LessOrEqualsTo",
+        ComparisonComparators.Equal: "Equals",
+        ComparisonComparators.NotEqual: "NotEquals",
+        ComparisonComparators.Like: "ContainsIgnoreCase",
+        ComparisonComparators.In: "Equals",
+        ComparisonComparators.Matches: 'ContainsIgnoreCase',
+        ObservationOperators.Or: '',  # 'OR' Not supported, error will be generated
+        ObservationOperators.And: 'AND'
+    }
+
+    def __init__(self, pattern: Pattern, data_model_mapper, options):
+
+        logger.info("Cybereason Connector")
+        self.dmm = data_model_mapper
```

Comment:
add this line `self.comparator_lookup = self.dmm.map_comparator()`

Reply:
Updated the code as suggested.